packages/server-api/src/index.ts

```diff
@@ -89,3 +89,75 @@ export interface Plugin {
   signalKApiRoutes?: (router: IRouter) => IRouter
   enabledByDefault?: boolean
 }
+
+export type DeltaInputHandler = (
+  delta: object,
+  next: (delta: object) => void
+) => void
+
+export interface Ports {
+  byId: string[]
+  byPath: string[]
+  byOpenPlotter: string[]
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  serialports: any
+}
+
+export interface ServerAPI extends PluginServerApp {
+  getSelfPath: (path: string) => void
+  getPath: (path: string) => void
+  getMetadata: (path: string) => void
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  putSelfPath: (aPath: string, value: any, updateCb: () => void) => Promise<any>
+  putPath: (
+    aPath: string,
+    value: number | string | object | boolean,
+    updateCb: (err?: Error) => void,
+    source: string
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  ) => Promise<any>
+  //TSTODO convert queryRequest to ts
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  queryRequest: (requestId: string) => Promise<any>
+  error: (msg: string) => void
+  debug: (msg: string) => void
+  registerDeltaInputHandler: (handler: DeltaInputHandler) => void
+  setProviderStatus: (msg: string) => void
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  handleMessage: (id: string, msg: any, skVersion?: SKVersion) => void
+  setProviderError: (msg: string) => void
```

Comment:
Hasn't `setProviderStatus` & `setProviderError`been deprecated and replaced by `setPluginStatus` and `setPluginError`?
Or have I got that the wrong way around?

Reply:
No, that's correct. Force pushed a fix.